lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -991,6 +991,11 @@ unsigned SharedEncodingAttr::getTotalElemsPerThread(ArrayRef<int64_t> shape,
   llvm_unreachable("getElemsPerThread is not supported for shared layout");
   return 0;
 }
+int32_t SharedEncodingAttr::getAlignment(Type eltTy) const {
+  if (getHasLeadingOffset())
+    return 128 * getMaxPhase();
+  return std::max(eltTy.getIntOrFloatBitWidth() * getVec(), 8u);
```

Comment:
Let me know if 16 is better. I think `size(type) * vec` is a principled choice, and 8 is the previous heuristic so we don't reduce alignment here.

Reply:
I believe in some cases we use a larger vector size that `vec` to do copies into shared memory. I'm surprised an alignment of 8 always works. Doing a 16bytes copy into an address non aligned on 16bytes would fail

Reply:
Changed default alignment to 16